ui/src/main/webapp/app/options/options-component.js

```diff
@@ -103,27 +102,14 @@ function OptionsController(operationOptions, config, events, $q, query, error) {
         }
     }
 
-    /**
-     * Unsubscribes from the event service to avoid too many event subscriptions.
-     */
-    vm.$onDestroy = function() {
-        events.unsubscribe('onPreExecute', saveToService);
-        saveToService()
-    }
-
-    var saveToService = function() {
-        if (vm.master) {        // If master is being destroyed, for example when the user navigates away, the service is updated
-            operationOptions.setDefaultConfiguration(vm.model);
-        }
-    }
-
     /**
      * Sets the value of an operation option to undefined if it is a string or empty array if already an array.
      * @param {Number} index The index of the option in the visible array
      */
     vm.clearValue = function(index) {
         var currentValue = vm.model.visible[index].value
         vm.model.visible[index].value = Array.isArray(currentValue) ? [] : undefined;
+        vm.onChange()
```

Comment:
Do you need to call onChange() from this method directly? Won't this be activated from the ng-change directive in the index.html